### PR TITLE
Fix `InputGroup.message=<StatusMessage>` rendering two icons

### PR DESCRIPTION
**Pull request by rohan-kadkol**
## Changes

After-PR TODO from #1753 

Earlier, `message` was always wrapped in `<StatusMessage>`. This caused nested `<StatusMessage>`s when `message={<StatusMessage>}`. Nested StatusMessages could lead to double svgIcons:

<details>
<summary>Example code</summary>

```jsx
{/* ❌ Passing <StatusMessage startIcon={…} /> to the message of <InputGroup status="…" /> shows duplicate status icons */}
<InputGroup
  label='Checkbox group'
  status='positive'
  message={
    <StatusMessage startIcon={<SvgStar />} status='positive'>
      Help message
    </StatusMessage>
  }
>
  <Checkbox label='Select all' />
  <Checkbox label='Football' />
  <Checkbox label='Hockey' />
</InputGroup>
```

</details>

image

This PR now wraps `message` in `<StatusMessage>` only when `typeof message !== "string"`. This fixes the duplicate svgIcons issue.

image

As a result of this change, if users were passing `ReactNode`s that are not `<StatusMessage>` to `message`, they might need to manually wrap their `ReactNode` with `<StatusMessage>` for proper styling of `message`.

<details>
<summary>Example code</summary>

```jsx
<InputGroup
  label='some label'
  message={<div className='my-message'>Message</div>}
>
  <Radio />
  <Radio />
  <Radio />
</InputGroup>
```
</details>

|Previously|Now|
|-|-|
|image|image|

## Testing

* Updated unit test to confirm that `message` is not wrapped in `<StatusMessage>` when `typeof message !== 'string'`.
* Fixed failing tests

## Docs

Updated JSDocs of `message` to tell that `message` is automatically wrapped with `<StatusMessage>` when `typeof message === 'string'`. It also tells that users may need to wrap their non-string & non-StatusMessage `message` with `<StatusMessage>` for proper styling.

### CODE REVIEW

**Review by Ben-Pusey-Bentley — APPROVED** [review]
Looks good to me.

File: packages/itwinui-react/src/core/InputGroup/InputGroup.tsx [hunk 1]
```diff
@@ -121,18 +125,29 @@ export const InputGroup = React.forwardRef((props, forwardedRef) => {
...
-            'aria-hidden': true,
-          }}
-          startIcon={svgIcon}
-          status={status}
-          {...messageProps}
-        >
-          {displayStyle !== 'inline' && message}
-        </StatusMessage>
-      )}
+      {(() => {
```
Comment (mayank99) [inline packages/itwinui-react/src/core/InputGroup/InputGroup.tsx:128]: Probably would be better to move this into a different component (in the same file), both for readability and performance

Comment (rohan-kadkol) [reply packages/itwinui-react/src/core/InputGroup/InputGroup.tsx]: Sounds good, done (https://github.com/iTwin/iTwinUI/pull/1877/commits/d5e0d4173d348e04631a22ebe6cdbcfa7eeef3bf).

Comment (mayank99) [reply packages/itwinui-react/src/core/InputGroup/InputGroup.tsx]: the curlies `{}` can also go.

**Review by mayank99 — APPROVED** [review]

**Pull request merged**